src/oge/data_cleaning.py

```diff
@@ -946,12 +977,81 @@ def assign_fuel_type_to_cems(cems, year, primary_fuel_table):
         filler_column="energy_source_code_epa",
     )
 
+    # if we are still missing fuel codes, use energy_source_code_1 of generator with
+    # greatest nameplate capacity
+    plant_backstop_fuel = assign_fuel_type_via_generator_with_greatest_capacity()
+    cems = cems.merge(
+        plant_backstop_fuel,
+        how="left",
+        on=["plant_id_eia"],
+        validate="m:1",
+    )
+    cems = fillna_with_missing_strings(
+        cems,
+        column_to_fill="energy_source_code",
+        filler_column="energy_source_code_plant",
+    )
+
     # update
     cems = update_energy_source_codes(cems, year)
 
     return cems
 
 
+def assign_fuel_type_via_generator_with_greatest_capacity() -> pd.DataFrame:
```

Comment:
I might suggest renaming this something like `load_backstop_energy_source_codes_for_plant()`. Otherwise looks good

Reply:
Done